solana-ibc-story.md

```diff
@@ -255,75 +256,75 @@ pub fn process_instruction(
 }
 ```
 
-It does is allocate 40 KB buffer which is larger than default heap
-size.  Therefore executing it as is in a transaction fails.  However,
-when executing a transaction consisting of two instructions: the first
+What this does is allocate a 40 KB buffer, which is larger than the default heap
+size.  Therefore, executing it in a transaction fails.  However,
+when executing a transaction consisting of two instructions, the first
 calling the Compute Budget program with `RequestHeapFrame(102400)`
 operation and the second calling the above test program, the program…
 fails just as before.
 
-Even though higher heap is requested, Solana program will have access
-to the first 32 KiB only.  Well, sort of.  Turns out the default
+Even though a higher heap is requested, the Solana program will have access
+to the first 32 KiB only.  Well, sort of.  It turns out the default
 allocator *assumes* that the heap is 32 KiB.  Even though the virtual
 machine allots more space, [the default allocator will use the first
 32 KiB
 only](https://docs.rs/solana-program/1.18.9/src/solana_program/entrypoint.rs.html#168).
-In `BumpAllocator` presented above this corresponds to the use of
+In the `BumpAllocator` presented above, this corresponds to the use of the
 `HEAP_LENGTH` constant.
 
 
 ### <a name="getting-heap-size"></a> Getting heap size
 
-That shouldn’t be a big problem.  Right?  Simply figure out the size
+Getting the heap size shouldn’t be a big problem.  Right?  One would hope to simply figure out the size
```

Comment:
Here the ‘that’ referred to default allocator assuming the size. Perhaps this would be better as something like:

```suggestion
Eliminating the hard-coded heap size shouldn’t be a big problem.  Right?  One would hope to simply figure out the size
```